crates/biome_js_analyze/src/analyzers/nursery/no_approximative_numeric_constant.rs

```diff
@@ -0,0 +1,112 @@
+    ///
+    /// Source: https://rust-lang.github.io/rust-clippy/master/#approx_constant
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```js,expect_diagnostic
+    /// let x = 3.141;
+    /// ```
+    /// ```js,expect_diagnostic
+    /// let x = 2.302;
+    /// ```
+    ///
+    /// ## Valid
+    ///
+    /// ```js
+    /// let x = Math.PI;
+    /// ```
+    /// ```js
+    /// let x = Math.LN10;
+    /// ```
+    ///
+    pub(crate) NoApproximativeNumericConstant {
+        version: "next",
+        name: "noApproximativeNumericConstant",
+        recommended: false,
+    }
+}
+
+impl Rule for NoApproximativeNumericConstant {
+    type Query = Ast<JsNumberLiteralExpression>;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if get_approximative_literal_diagnostic(node).is_some() {
+            Some(())
+        } else {
+            None
+        }
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        get_approximative_literal_diagnostic(node)
+    }
+}
+
+// Tuples are of the form (constant, name, min_digits)
+const KNOWN_CONSTS: [(f64, &str, usize); 8] = [
+    (f64::E, "E", 4),
+    (f64::LN_10, "LN10", 4),
+    (f64::LN_2, "LN2", 4),
+    (f64::LOG10_E, "LOG10E", 4),
+    (f64::LOG2_E, "LOG2E", 4),
+    (f64::PI, "PI", 4),
+    (f64::FRAC_1_SQRT_2, "SQRT1_2", 4),
+    (f64::SQRT_2, "SQRT2", 4),
+];
+
+fn get_approximative_literal_diagnostic(
+    node: &JsNumberLiteralExpression,
+) -> Option<RuleDiagnostic> {
+    let binding = node.text();
+    let s = binding.trim();
+    if s.parse::<f64>().is_err() {
+        return None;
+    }
+
+    for &(constant, name, min_digits) in &KNOWN_CONSTS {
+        if is_approx_const(constant, s, min_digits) {
+            return Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    node.syntax().text_trimmed_range(),
+                    markup! { "Prefer constants from the standard library." },
+                )
+                .note(markup! { "Use "<Emphasis>"Math."{ name }</Emphasis>" instead." }),
+            );
+        }
+    }
+
+    None
+}
+
+/// Returns `false` if the number of significant figures in `value` are
+/// less than `min_digits`; otherwise, returns true if `value` is equal
+/// to `constant`, rounded to the number of digits present in `value`.
+fn is_approx_const(constant: f64, value: &str, min_digits: usize) -> bool {
+    if value.len() <= min_digits {
+        false
+    } else if constant.to_string().starts_with(value) {
+        // The value is a truncated constant
+        true
+    } else {
+        let round_const = format!("{constant:.*}", value.len() - 2);
+        value == round_const
+    }
+}
```

Comment:
I have to admin that I copied this function from the clippy linter in the original issue:
https://github.com/rust-lang/rust-clippy/blob/9554e477c29e6ddca9e5cdce71524341ef9d48e8/clippy_lints/src/approx_const.rs#L118-L132

I dont know whether using f64 for comparisons would work, since floats cannot accurately represent every number.

What would work, however, would be an heuristic that we only check this (and do the string allocation) if the derivation of the float is less than `0.1` (or some other limit). This could avoid most of the allocations.

Reply:
Not a big issue! We can keep it as is :) 

Maybe let's put a link to the original code in the issue, I think it's fair to credit them

Reply:
Yeah, sure